### PR TITLE
merger(ticdc): reduce sorter merger goroutine

### PR DESCRIPTION
**Pull request by sdojjy**
### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tiflow/issues/4351

### What is changed and how it works?
1. refactor merger function into a struct Merger.
2. split the merge function to two functions, one consume flush task non-blocking, another send ordered events to next node.
3. spawns goroutine and wait in the sorter

### Check List 

Tests 

 - Unit test
 - Integration test

### Release note 

```release-note
 `None`.
```

### CODE REVIEW

File: cdc/sorter/unified/merger.go [hunk 1]
```diff
@@ -404,18 +403,40 @@ func runMerger(ctx context.Context, numSorters int, in <-chan *flushTask, out ch
...
+			return ctx.Err()
+		case <-resolvedTsTicker.C:
+			if err := resolvedTsTickFunc(); err != nil {
+				return err
+			}
+		case <-resolvedTsTickCh:
+			if err := resolvedTsTickFunc(); err != nil {
+				return err
+			}
+		case task = <-in:
```
Comment (liuzix) [inline cdc/sorter/unified/merger.go:439]: This will reintroduce the problem described in https://github.com/pingcap/tiflow/pull/1574

Comment (sdojjy) [reply cdc/sorter/unified/merger.go:451]: fixed

File: cdc/sorter/unified/merger.go [hunk 2]
```diff
@@ -404,89 +405,83 @@ func runMerger(ctx context.Context, numSorters int, in <-chan *flushTask, out ch
...
+}
 
-			if minTemp > minResolvedTs {
-				atomic.StoreUint64(&minResolvedTs, minTemp)
-				select {
-				case resolvedTsNotifierChan <- struct{}{}:
-				default:
-				}
-			}
+func (m *sorterMerger) runFlushTask(ctx context.Context, numSorters int, in <-chan *flushTask) error {
```
Comment (liuzix) [inline cdc/sorter/unified/merger.go:444]: Logically `numSorters` is a property of the `sorterMerger`. Can you put `numSorters` inside `sorterMerger`?

File: cdc/sorter/unified/unified_sorter.go [hunk 3]
```diff
@@ -170,8 +170,16 @@ func (s *Sorter) Run(ctx context.Context) error {
 		}
 	})
 
+	merger := &sorterMerger{
```
Comment (liuzix) [inline cdc/sorter/unified/unified_sorter.go:173]: Maybe we need a constructor function here?

File: cdc/sorter/unified/merger.go (hunk 2 above)
Comment (sdojjy) [reply cdc/sorter/unified/merger.go:444]: done

File: cdc/sorter/unified/unified_sorter.go (hunk 3 above)
Comment (sdojjy) [reply cdc/sorter/unified/unified_sorter.go]: done

**Pull request closed without merging**